build/builder.go

```diff
@@ -454,6 +454,7 @@ func (b *Builder) deleteRemainingShards() {
 		if err := os.Remove(name); os.IsNotExist(err) {
 			break
 		} else {
+			_ = os.Remove(name + ".meta")
```

Comment:
Could we always remove the metadata file, even when the shard has been already deleted? I think that would be more robust to certain partial failure scenarios.

Reply:
We use the shard missing as a sign to stop looking for more. ie they are incrementally assigned numbers starting at `00000`. We don't do something like `glob` to find errant shards. So this is the correct place to stop.

However, I agree with the sentiment. Will add a global cleanup which ensures this property (like we do for cleaning up stale tmp files).